tensorflow/core/common_runtime/gpu/gpu_cudamallocasync_allocator.cc

```diff
@@ -214,10 +218,57 @@ void* GpuCudaMallocAsyncAllocator::AllocateRaw(size_t alignment,
     cuMemGetInfo(&free, &total);
     mutex_lock lock(lock_);
     LOG(ERROR) << Name() << " cuMemAllocAsync failed to allocate " << num_bytes
-               << ": " << GetCudaErrorMessage(result)
-               << "\n Free memory/Total memory: " << free << "/" << total;
+               << " bytes: " << GetCudaErrorMessage(result)
+               << "\n Reported by CUDA: Free memory/Total memory: " << free << "/" << total;
     if (auto stats = GetStats())
       LOG(ERROR) << "Stats: " << stats->DebugString();
+
+    std::map<size_t, int> size_map_historgram;
```

Comment:
Let's pull all of this out into a function with a clear & obvious name.

Reply:
done.